utils/writeable/src/impls.rs

```diff
@@ -53,55 +50,62 @@ macro_rules! impl_write_num {
-                    LengthHint::exact(high as usize)
+                    $crate::LengthHint::exact(high as usize)
                 }
             }
         }
 
-        impl Writeable for $i {
-            fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
+        impl $crate::Writeable for $i {
+            fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
                 if self.is_negative() {
                     sink.write_str("-")?;
                 }
                 self.unsigned_abs().write_to(sink)
             }
 
-            fn write_len(&self) -> LengthHint {
-                LengthHint::exact(if self.is_negative() { 1 } else { 0 })
+            fn write_len(&self) -> $crate::LengthHint {
+                $crate::LengthHint::exact(if self.is_negative() { 1 } else { 0 })
                     + self.unsigned_abs().write_len()
             }
         }
-    };
-}
 
-impl_write_num!(u8, i8);
-impl_write_num!(u16, i16);
-impl_write_num!(u32, i32);
-impl_write_num!(u64, i64);
-impl_write_num!(u128, i128);
+        #[test]
+        fn $test() {
+            use $crate::assert_writeable_eq;
+            assert_writeable_eq!(&(0 as $u), "0");
+            assert_writeable_eq!(&(0 as $u), "0");
+            assert_writeable_eq!(&(-0 as $i), "0");
+            assert_writeable_eq!(&(1 as $u), "1");
+            assert_writeable_eq!(&(1 as $i), "1");
+            assert_writeable_eq!(&(-1 as $i), "-1");
+            assert_writeable_eq!(&(10 as $u), "10");
+            assert_writeable_eq!(&(10 as $i), "10");
+            assert_writeable_eq!(&(-10 as $i), "-10");
+            assert_writeable_eq!(&(99 as $u), "99");
+            assert_writeable_eq!(&(99 as $i), "99");
+            assert_writeable_eq!(&(-99 as $i), "-99");
+            assert_writeable_eq!(&(100 as $u), "100");
+            assert_writeable_eq!(&(-100 as $i), "-100");
+            assert_writeable_eq!(&<$u>::MAX, <$u>::MAX.to_string());
+            assert_writeable_eq!(&<$i>::MAX, <$i>::MAX.to_string());
+            assert_writeable_eq!(&<$i>::MIN, <$i>::MIN.to_string());
 
-#[test]
-fn test_ints() {
-    use crate::assert_writeable_eq;
-    assert_writeable_eq!(&0u128, "0");
-    assert_writeable_eq!(&0i128, "0");
-    assert_writeable_eq!(&-0i128, "0");
-    assert_writeable_eq!(&1u128, "1");
-    assert_writeable_eq!(&1i128, "1");
-    assert_writeable_eq!(&-1i128, "-1");
-    assert_writeable_eq!(&10u128, "10");
-    assert_writeable_eq!(&10i128, "10");
-    assert_writeable_eq!(&-10i128, "-10");
-    assert_writeable_eq!(&99u128, "99");
-    assert_writeable_eq!(&99i128, "99");
-    assert_writeable_eq!(&-99i128, "-99");
-    assert_writeable_eq!(&u128::MAX, u128::MAX.to_string());
-    assert_writeable_eq!(&i128::MAX, i128::MAX.to_string());
-    assert_writeable_eq!(&i128::MIN, i128::MIN.to_string());
+            for _ in 0..1000 {
+                let rand = rand::random::<$u>();
```

Comment:
On the one hand, we want to fuzz, but on the other hand, we don't want our tests to be nondeterministic.  I think we should probably go safe and seed the PRNG.